### PR TITLE
Make UI look-and-feel consistent #299

### LINKED ISSUE #299: Make UI look-and-feel consistent
The UI of the [original addressbook](http://code.makery.ch/library/javafx-8-tutorial/) looks prettier. We should probably ensure all elements use the same theme so that the entire UI looks consistent.

### PR DESCRIPTION
**Pull request by MightyCupcakes**
Closes #299

Before:

capture1

After:

capture

Proposed merge commit message:

```
The UI contains multiple elements with contrasting white background.

This is not desirable as it disrupts the general look-and-feel of the
dark themed user interface with several big white boxes, making the
interface look very clunky.

Let's modify these elements so that they fit with the general dark 
theme.
```

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@MightyCupcakes submitted v1 for review.

* [v1 1/3] [StatusBarFooter: Change save location status to align to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/21520092dbd29011a838aed80696d91afae92c9a)
* [v1 2/3] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/568d5f4115a40d89b8f14a425f996278a702ab5e)
* [v1 3/3] [BrowserPanel: Modify browser to load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/f080f2250dcace2f2b2e85ac2c387f468276b56d)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/65fdb4818f191c27f46846588172f644483718af...f080f2250dcace2f2b2e85ac2c387f468276b56d))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by damithc** [comment]
How does it look without card outlines? Or more subtle outlines?

**Comment by damithc** [comment]
Also note how in the original addressbook the status bar has a different shade of black from the rest.

**Comment by MightyCupcakes** [comment]
@damithc 

No borders:

capture2

Subtle borders:

capture3

**Comment by damithc** [comment]
To me, these two look cleaner than the first one. May be can get rid of the two vertical borders near the scroll bar in the middle? Would also like to see a screenshot with a webpage loaded and a card selected.

**Comment by yl-coder** [comment]
@chao1995 I think you can switch color scheme by setting the styleClass programmatically.

**Comment by MightyCupcakes** [comment]
@chao1995 

As the CSS file is named `DarkTheme.css`, I don't think a white theme works here. We could maintain two separate CSS files and allow the user to specify his/her preferences but that is way outside the scope of this PR.

**Comment by damithc** [comment]
I don't think we can get a 100% agreement on UI design matters. I'm fine with the design that has majority consensus (unless I decide to veto :-p). 
Switching themes programmatically is something we can do, but let's leave it as a future 'nice to have' enhancement. For the moment, a dark flat theme seems the easiest to achieve. But I worry that it'll look ugly when the webpage is loaded.
I'm not against switching to a different/light scheme altogether (we can change the css file name), if we can find one that looks just as modern as the current one. I don't think we have the expertise to create a totally new professional-looking theme ourselves :-p

**Comment by CanIHasReview** [comment]
# v2

@MightyCupcakes submitted v2 for review.

* [v2 1/3] [StatusBarFooter: Change save location status to align to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/21520092dbd29011a838aed80696d91afae92c9a)
* [v2 2/3] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/6ca60238756b01d3f177cad3dd2551780baabe00)
* [v2 3/3] [BrowserPanel: Modify browser to load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/3ac9d8f4940035aafcabad4c5ab4789d83c539e0)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/65fdb4818f191c27f46846588172f644483718af...3ac9d8f4940035aafcabad4c5ab4789d83c539e0)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/757e396f06cd124d66f06b03950fd10c7d3dc2a1~1...757e396f06cd124d66f06b03950fd10c7d3dc2a1))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by MightyCupcakes** [comment]
Made several changes to the CSS files.

New look (no card borders):

capture3

Person selected:
capture2

Invalid command:
capture4


## Known issues
1. The annoying square in the `WebView` cannot be styled despite my best efforts. Yes, I have tried using the `corner`, `viewport` class and all its known variants. [It seems it is not part of Javafx controls](http://stackoverflow.com/questions/15498287/remove-the-scrollbar-in-the-webview-javafx).

1. The use of `!important` in `.error` class. This is needed (when it is not needed previously) because of the fact that there is custom styling for both background colour and text colour. Without `!important`, the style will remain unchanged despite the fact that the `error` class is correctly added to the command box.

File: src/main/resources/view/blank.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!doctype html>
+<html>
+<head>
+     <link rel="stylesheet" href="DarkTheme.css">
+</head>
+
+<body class = "background">
```
Comment (yamgent) [inline src/main/resources/view/blank.html]: No need to leave spaces (just `class="background"`).

File: src/main/resources/view/DarkTheme.css [hunk 2]
```diff
@@ -86,23 +87,42 @@
...
+
+.list-cell:filled:selected:focused {
+    -fx-background-color: #424d5f;
+}
+
+.list-cell:filled:even {
+    -fx-background-color: #3c3e3f;
+}
+
+.list-cell:filled:odd {
```
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: `.list-cell:pseudo-class` before `.list-cell .label`, which targets `.label`.

**Comment by MightyCupcakes** [comment]
@chao1995 any comments for this iteration?

**Comment by CanIHasReview** [comment]
# v3

@MightyCupcakes submitted v3 for review.

* [v3 1/4] [StatusBarFooter: Align save location to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/37653b6c7fce02d7dbc949c09c2fe678867ecc7a)
* [v3 2/4] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/5ce7f313abc9c672046a1a8b1eb2e93d0c5ab025)
* [v3 3/4] [BrowserPanel: Load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/9dcf989315f5b226c06c7861419e74e25199a46f)
* [v3 4/4] [Update screenshot of GUI in docs ](https://github.com/se-edu/addressbook-level4/pull/314/commits/dea8a39fe82bbb31d0d77d2e00456a63c53dc91b)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/65fdb4818f191c27f46846588172f644483718af...dea8a39fe82bbb31d0d77d2e00456a63c53dc91b)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/2cb383ca1f33e733bc8aecffee678bea1004d262~1...2cb383ca1f33e733bc8aecffee678bea1004d262))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by MightyCupcakes** [comment]
Added a border for selected `PersonCard`

capture2

**Comment by MightyCupcakes** [comment]
@yamgent 

The annoying square in the `WebView` is a known issue and cannot be styled despite my best efforts. [It seems it is not part of Javafx controls](http://stackoverflow.com/questions/15498287/remove-the-scrollbar-in-the-webview-javafx).

**Comment by yamgent** [comment]
Oh sorry, I just realised you addressed it in an earlier post. That problem is unfortunate.

**Comment by MightyCupcakes** [comment]
@chao1995 comments?

File: src/main/resources/view/blank.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
```
Comment (damithc) [inline src/main/resources/view/blank.html]: We may want to add some content to this page in future, right? e.g. a welcome message or a daily quote 
In that case, not name it as `blank.html`

**Comment by CanIHasReview** [comment]
# v4

@MightyCupcakes submitted v4 for review.

* [v4 1/4] [StatusBarFooter: Align save location to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/1f04601ae2b4f99ddd059d9ccf4d765a5f58dcd0)
* [v4 2/4] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/e482755345fddb245a1c3c29db75b209a7eb22e1)
* [v4 3/4] [BrowserPanel: Load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/711d077d4d4fbb348377bb0ab746be9f1f638d59)
* [v4 4/4] [Update screenshot of GUI in docs ](https://github.com/se-edu/addressbook-level4/pull/314/commits/eca651737468055930a95d7ac56e1048d7d15df1)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/297506b87fabe6de6baa7901388039ec50199802...eca651737468055930a95d7ac56e1048d7d15df1)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/9b2a44e5c6cf53ff5fe4243bc7b0b50564c40b58~1...9b2a44e5c6cf53ff5fe4243bc7b0b50564c40b58))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by damithc** [comment]
@acjh for your review (in case you didn't get a notification)

File: src/main/resources/view/StatusBarFooter.fxml [hunk 4]
```diff
@@ -9,6 +9,6 @@
 </columnConstraints>
  <children>
       <StatusBar styleClass="anchor-pane" fx:id="syncStatus" minWidth="0.0"/>
-      <StatusBar styleClass="anchor-pane" fx:id="saveLocationStatus" minWidth="0.0" GridPane.columnIndex="1"/>
+      <StatusBar styleClass="anchor-pane" fx:id="saveLocationStatus" minWidth="0.0" GridPane.columnIndex="1" nodeOrientation="RIGHT_TO_LEFT" />
```
Comment (damithc) [inline src/main/resources/view/StatusBarFooter.fxml]: This line too long?

**Comment by damithc** [comment]
@yl-coder anything to add? CC: @pyokagan

**Comment by CanIHasReview** [comment]
# v5

@MightyCupcakes submitted v5 for review.

* [v5 1/4] [StatusBarFooter: Align save location to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/977f98ec52f81408625db5c0f42543f03e970f8b)
* [v5 2/4] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/c52a4c6eaf2753015f1315f37bbf133304f3a2af)
* [v5 3/4] [BrowserPanel: Load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/c5a0ea7b26e4c57fcc089557a379b1cdc75ae1ad)
* [v5 4/4] [Update screenshot of GUI in docs ](https://github.com/se-edu/addressbook-level4/pull/314/commits/4121ec83eaf23d1af6dcc5269125f96e10d84c0a)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/297506b87fabe6de6baa7901388039ec50199802...4121ec83eaf23d1af6dcc5269125f96e10d84c0a)) ([:chart_with_upwards_trend: Interdiff between v4 and v5](https://github.com/se-edu/addressbook-level4/compare/c12b7471762d5626786932ba3c30eca12124f5b7~1...c12b7471762d5626786932ba3c30eca12124f5b7))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/5/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: src/main/java/seedu/address/ui/BrowserPanel.java [hunk 5]
```diff
@@ -38,6 +44,15 @@ public void loadPage(String url) {
     }
 
     /**
+     * Loads an default HTML file with a matching background on start-up
```
Comment (acjh) [inline src/main/java/seedu/address/ui/BrowserPanel.java]: How about:
`Loads a default HTML file with a background that matches the general theme.`

The function itself does not load it on startup.

File: src/main/resources/view/DarkTheme.css [hunk 6]
```diff
@@ -1,5 +1,6 @@
 .background {
     -fx-background-color: derive(#1d1d1d, 20%);
+    background-color: #383838; /* Used in the blank.html file */
```
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: "Used in the `default`.html file"

File: src/main/resources/view/DarkTheme.css [hunk 7]
```diff
@@ -274,27 +307,42 @@
...
 #commandTypeLabel {
     -fx-font-size: 11px;
     -fx-text-fill: #F70D1A;
 }
 
+#commandTextField {
+    -fx-background-insets: 0;
+    -fx-background-color: transparent #383838 transparent #383838;
+    -fx-border-insets: 0;
+    -fx-border-width: 1px;
```
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: `-fx-border-width` doesn't use `px`

File: src/main/resources/view/default.html [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+<head>
+     <link rel="stylesheet" href="DarkTheme.css">
```
Comment (acjh) [inline src/main/resources/view/default.html]: Indentation should be 4 spaces, not 5.

File: src/main/resources/view/DarkTheme.css (hunk 7 above)
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: Also, `-color` before `-insets`

File: src/main/resources/view/DarkTheme.css [hunk 9]
```diff
@@ -274,27 +307,42 @@
...
     -fx-font-size: 11px;
     -fx-text-fill: #F70D1A;
 }
 
+#commandTextField {
+    -fx-background-insets: 0;
+    -fx-background-color: transparent #383838 transparent #383838;
+    -fx-border-insets: 0;
+    -fx-border-width: 1;
+    -fx-border-color: #383838 #383838 #ffffff #383838;
```
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: Alphabetical order for the 5 properties above.

File: src/main/resources/view/DarkTheme.css [hunk 10]
```diff
@@ -76,8 +77,8 @@
 }
 
 .split-pane:horizontal .split-pane-divider {
-    -fx-border-color: transparent #1d1d1d transparent #1d1d1d;
-    -fx-background-color: transparent, derive(#1d1d1d, 10%);
+    -fx-border-color: transparent transparent transparent #4d4d4d;
+    -fx-background-color: derive(#1d1d1d, 20%);
```
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: `-background` before `-border`

**Comment by CanIHasReview** [comment]
# v6

@MightyCupcakes submitted v6 for review.

* [v6 1/4] [StatusBarFooter: Align save location to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/977f98ec52f81408625db5c0f42543f03e970f8b)
* [v6 2/4] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/b4e29ddd15955e1bcc8b41a1398f4299bf96a168)
* [v6 3/4] [BrowserPanel: Load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/d8fd23778fbc1ae4bb18ed946ac53956e81ebc74)
* [v6 4/4] [Update screenshot of GUI in docs ](https://github.com/se-edu/addressbook-level4/pull/314/commits/6cf4caf676e72718af84cf7c087fdf9d2c4e1777)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/297506b87fabe6de6baa7901388039ec50199802...6cf4caf676e72718af84cf7c087fdf9d2c4e1777)) ([:chart_with_upwards_trend: Interdiff between v5 and v6](https://github.com/se-edu/addressbook-level4/compare/3c428c55ec3171a1aa738f29c1c249e60add1f0e~1...3c428c55ec3171a1aa738f29c1c249e60add1f0e))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/6/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: src/main/resources/view/DarkTheme.css [hunk 11]
```diff
@@ -1,5 +1,6 @@
 .background {
     -fx-background-color: derive(#1d1d1d, 20%);
+    background-color: #383838; /* Used in the default html file */
```
Comment (acjh) [inline src/main/resources/view/DarkTheme.css]: default`.`html?

**Comment by CanIHasReview** [comment]
# v7

@MightyCupcakes submitted v7 for review.

* [v7 1/4] [StatusBarFooter: Align save location to the right](https://github.com/se-edu/addressbook-level4/pull/314/commits/977f98ec52f81408625db5c0f42543f03e970f8b)
* [v7 2/4] [Modify CSS files to better match the general dark theme](https://github.com/se-edu/addressbook-level4/pull/314/commits/104fad3b3c777336e6929c7cd624c574259c1032)
* [v7 3/4] [BrowserPanel: Load a default blank page](https://github.com/se-edu/addressbook-level4/pull/314/commits/79fb9c3f15a9323a7bb6939616ce28570fd1dcc6)
* [v7 4/4] [Update screenshot of GUI in docs ](https://github.com/se-edu/addressbook-level4/pull/314/commits/53e85de25ff2d1c91e7312583dff101275807180)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/297506b87fabe6de6baa7901388039ec50199802...53e85de25ff2d1c91e7312583dff101275807180)) ([:chart_with_upwards_trend: Interdiff between v6 and v7](https://github.com/se-edu/addressbook-level4/compare/db03b8361fee628246cc71431e96f59f3d5ffb70~1...db03b8361fee628246cc71431e96f59f3d5ffb70))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/314/7/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by yl-coder** [comment]
Looks good. Just take note of the difference between px and pt and use them accordingly.

**Comment by damithc** [comment]
@yamgent @chao1995 @acjh please review changes done since your last approval.

**Comment by damithc** [comment]
@MightyCupcakes sorry I missed this one. Can you propose a merge commit message? Add it to the PR description.

**Comment by damithc** [comment]
Will be merging soon
CC: @pyokagan

**Comment by damithc** [comment]
@yamgent @MightyCupcakes @chao1995 please update your week 9 progress report by today if you want to include this PR for week 9.

**Pull request merged**